gamemode/modules/base/cl_entityvars.lua

```diff
@@ -8,10 +8,11 @@ local pmeta = FindMetaTable("Player")
 -- enough to warrant optimizing. See https://github.com/FPtje/DarkRP/pull/3212
 local get_user_id = pmeta.UserID
 function pmeta:getDarkRPVar(var, fallback)
-    fallback = fallback or nil
+    fallback = fallback ~= nil and fallback or nil
```

Comment:
This doesn't fix it. ``true and false or nil`` evaluates to nil. Copy the == nil check idiom on line 13.

Reply:
I would just cut that check out and make a check at the bottom for if results == nil and fallback ~= nil. Seems like the better option, if I am not mistaken

Reply:
```Lua
function pmeta:getDarkRPVar(var, fallback)
    local vars = DarkRPVars[get_user_id(self)]
    if vars == nil then return fallback end
    
    local results = vars[var]
    if results == nil then return fallback end
	
    return results
end
```
This is the cleanest imo